daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/World.scala

```diff
@@ -6,16 +6,18 @@ package com.daml.lf.validation
 import com.daml.lf.data.Ref._
 import com.daml.lf.language.Ast
 
-private[validation] class World(packages: PartialFunction[PackageId, Ast.Package]) {
+private[validation] class World(
+    signatures: PartialFunction[PackageId, Ast.AbstractPackage[_]]
```

Comment:
Yes. It is explained in the description of the PR. 
> The Ast validation requires indistinctively the full ASTs or the signatures of the dependencies of the package it has to check.
The reason is that I do not want to do extra work on a package on a package (to convert it into a signature) before it is validated. Additionally having to generate the signature for the validation, oblige we either do it twice (one for validation and one for compilation) or we have to tide more closely the two phases in order to reuse the same `PackageSignature` in both case (this is not completely straightforward in the ledger)

Reply:
That explanation makes sense. Thank you.

I assume we never hold on to objects of type `World` for long and thus there's no danger of inadvertently holding on to a full package instead of a signature. Is that correct?

Reply:
Yes.

Reply:
That's good to know. Merci.